server/src/main/java/org/apache/druid/segment/loading/SegmentLocalCacheManager.java

```diff
@@ -95,6 +104,14 @@ public SegmentLocalCacheManager(
     this.locations = locations;
     this.strategy = strategy;
     log.info("Using storage location strategy: [%s]", this.strategy.getClass().getSimpleName());
+
+    if (this.config.getNumThreadsToLoadSegmentsIntoPageCacheOnDownload() != 0) {
```

Comment:
Should add validation for negative values

Reply:
Without validation historical start up will fail with IllegalArgumentException, users will have to figure out the cause for it which is also fine.